### ISSUE TITLE
Providing an invalid file path should raise instead of silently ignoring it

### ISSUE THREAD

**Issue by jwoertink** [crystal-ameba/ameba#388]
Sort of related: https://github.com/crystal-ameba/ameba/issues/362

But in this case, if you just tell ameba to check a file that doesn't exist, it returns Inspecting 0 files

```
❯ ./bin/ameba sdfawsdf
Inspecting 0 files



Finished in 10 microseconds
0 inspected, 0 failures
```

I'm thinking of maybe a situation where someone has a CI and types the name of a path wrong, the CI runs and says things are all good, but you don't find out until later that's not the case.

**Comment by Sija** [comment]
Duplicate of #362

**Comment by veelenga** [comment]
@Sija this looks like a separate issue. See the difference between config file path and a file to check.

**Comment by jwoertink** [comment]
Yeah, I originally considered adding to the other issue, but thought maybe they'd be different enough where one is 

```
./bin/ameba -c not_there.yml
```

and this is

```
./bin/ameba not_there.cr
```

I guess another option could be to combine them in to a single issue that's like "passing file paths that don't exist should raise an error in all cases"?

**Comment by Sija** [comment]
You're right, I was trigger happy 😅

**Comment by stufro** [comment]
This is an interesting one to implement because the command line arguments can be a single file or a glob. In [glob_utils.cr#expand](https://github.com/crystal-ameba/ameba/blob/master/src/ameba/glob_utils.cr#L22-L27) the list of arguments are expanded and only file paths which exist are returned from `Dir[glob]`.

This makes it hard to catch non-existent files passed on the command line anywhere else in the code base after the globs have been expanded, which is why I'm thinking to implement this in `GlobUtils#expand`.

My idea is to raise an exception if a particular glob doesn't return any results from `Dir[glob]`. This would raise in both the below examples:

```
> ./bin/ameba not_there.cr
Error: No files found matching not_there.cr

> ./bin/ameba not_there/**/*.cr
Error: No files found matching not_there/**/*..cr
```

I'll make a PR using this approach, but I'm open to feedback on different approaches.

**Comment by veelenga** [comment]
@stufro thanks for taking a look.

This is what I would expect, however, need to understand if there are non-intuitive edge cases.

**Comment by stufro** [comment]
Thanks - #394. I've tried all the edge cases I can think of and it work as I would expect.
